lib/src/fake_aggregate_query.dart

```diff
@@ -28,10 +31,18 @@ class FakeAggregateQuery implements AggregateQuery {
   @override
   AggregateQuery count() => _query.count();
 
+  Map<String, dynamic> _getRawDocDataMap(QueryDocumentSnapshot<Object?> s) {
+    if (s is MockQueryDocumentSnapshot && s.snapshot is MockDocumentSnapshot) {
+      return (s.snapshot as MockDocumentSnapshot).getRawDocument();
+    }
+    // this line should never execute but leave it here to play safe.
+    return s.data() as Map<String, dynamic>;
+  }
+
   AggregateQuerySnapshotPlatform _getAggregateQuerySnapshotPlatform({
     required QuerySnapshot<Object?> snapshot,
   }) {
-    final dataMaps = snapshot.docs.map((e) => e.data() as Map<String, dynamic>);
+    final dataMaps = snapshot.docs.map((s) => _getRawDocDataMap(s));
```

Comment:
Have you considered using `DocumentSnapshot.get`? Then you don't need to expose the internal map. It's better to use the public API whenever possible.

You would replace
```dart
final value = dataMap[field.field];
```
by
```dart
final value = documentSnapshot.get(field.field);
```

It also has the added benefit to make the loop down there clearer. Instead of this:
```dart
    for (final dataMap in dataMaps) {
      for (final field in fields) {
```
you could write:
```dart
    for (final field in fields) {
      for (final documentSnapshot in documentSnapshots) {
```

Also would you mind renaming `valueMap` into something more meaningful like `aggregateValues`? Since you already use the variable name `value` when referring to a document value, `value` and `valueMap` was a bit confusing.

Reply:
Good idea!